iota-execution/latest/iota-adapter/src/execution_engine.rs

```diff
@@ -784,18 +785,19 @@ mod checked {
         mut builder: ProgrammableTransactionBuilder,
         params: &AdvanceEpochParams,
     ) -> Result<ProgrammableTransaction, ExecutionError> {
-        // Step 1: Create storage charges and computation rewards.
-        let (storage_charges, computation_rewards) = mint_epoch_rewards_in_pt(&mut builder, params);
+        // Step 1: Create storage and computation charges.
+        let (storage_charges, computation_charges) = mint_epoch_rewards_in_pt(&mut builder, params);
 
         // Step 2: Advance the epoch.
         let mut arguments = vec![
             builder
-                .pure(params.validator_target_reward)
+                .pure(params.validator_subsidy)
                 .expect("bcs encoding a u64 should not fail"),
             storage_charges,
-            computation_rewards,
+            computation_charges,
         ];
         let call_arg_arguments = vec![
+            CallArg::Pure(bcs::to_bytes(&params.computation_charge_burned).unwrap()),
```

Comment:
If it is the only change in the executor we can consider options instead of creating a new execution layer cut, for example:
1. Create a new transaction type `EndOfEpochTransactionKind::ChangeEpochV2`.
2. Or even easier, add a feature flag in the protocol config `advance_epoch_v2`, set it to `true` for the second protocol version, and add the `computation_charge_burned` parameter if it is enabled.

Reply:
I think too we can definitively try avoid the use of a new cut!

Reply:
Ok, if that is the preferred approach I can try one of the suggested alternatives.